key.core/src/main/java/de/uka/ilkd/key/nparser/ParsingFacade.java

```diff
@@ -128,21 +129,22 @@ public static KeyAst.File parseFile(File file) throws IOException {
 
     public static KeyAst.File parseFile(CharStream stream) {
         KeYParser p = createParser(stream);
-
+        var ep = p.getErrorReporter();
         p.getInterpreter().setPredictionMode(PredictionMode.SLL);
         // we don't want error messages or recovery during first try
-        p.removeErrorListeners();
-        p.setErrorHandler(new BailErrorStrategy());
-        KeYParser.FileContext ctx;
+        KeYParser.FileContext ctx = null;
         try {
             ctx = p.file();
-        } catch (ParseCancellationException ex) {
+        } catch (ParseCancellationException ex1) {
```

Comment:
Is it guaranteed to always be a ParseCancellationException?

Reply:
Yes. The `ParseCancellationException` is thrown from the `SyntaxErrorReporter` on recovering from parse errors. `ParseException`s are caught during the Parser, and given to the error listener and error strategy. 

It is a little bit complicated.